### DISCUSSION TITLE
Smaller package size?

### DISCUSSION THREAD

**Discussion by Benjythebee** [TanStack/query#8083] [General]
npmjs says the unpacked size of @tanstack/react-query is 1.1mb and I only need the useQuery hook in react. I'm not too keen on adding 1.1mb just for that single hook 😅 Is there an alternative or maybe a package that only exports useQuery?

**Comment by TkDodo** [comment]
what npm reports is what you install when you run `npm install`, NOT what you ship to the user. I’m a bit baffled that developers actually believe react-query ships 1.1MB of JavaScript for a single hook to the browser.

The full bundle size of react-query is [41.2kb minified, 12.1kb gzipped](https://bundlejs.com/?q=%40tanstack%2Freact-query&config=%7B%22esbuild%22%3A%7B%22external%22%3A%5B%22react%22%2C%22react-dom%22%5D%7D%7D).

If you only need `useQuery`, it’s [30.4kb minified, 9.14kb gzipped](https://bundlejs.com/?q=%40tanstack%2Freact-query&treeshake=%5B%7BuseQuery%2CQueryClient%2CQueryClientProvider+%7D%5D&config=%7B%22esbuild%22%3A%7B%22external%22%3A%5B%22react%22%2C%22react-dom%22%5D%7D%7D).

**Comment by Benjythebee** [comment]
`I’m a bit baffled that developers actually believe react-query ships 1.1MB`, you never know! 

Thanks a lot for the information though! Pretty sick
